(feat): Adr search collator module (#549)
backstage/community-plugins: workspaces/adr/.changeset/calm-mirrors-mate.md

---

> '@backstage-community/search-backend-module-adr': minor

Reviewer: @backstage/community-plugins-maintainers any guidance on whether search modules should reside within the main monorepo or with specific plugins here? eg. the explore search module still lives in the main repo: https://github.com/backstage/backstage/tree/master/plugins/search-backend-module-explore

Reviewer 2: I think it's fine to be here!

---

> Initial version for the search-backend-module-adr. This module can be used to
> add the search collator directly without having to create the backend module
> on your own if using the new backend system.

Reviewer 2: ```suggestion
Initial version for the `@backstage-community/search-backend-module-adr`. This module can be used to add the `adr` search collator using the new backend system.
```

Author: Reworded the changelog. Also added a second paragraph with the types that have been moved
